src/java/org/apache/cassandra/schema/PartitionDenylist.java

```diff
@@ -0,0 +1,462 @@
+        logger.info(retryReason + " while loading partition denylist cache.  Scheduled retry in {} seconds.",
+                    retryInSeconds);
+        ScheduledExecutors.optionalTasks.schedule(this::initialLoad, retryInSeconds, TimeUnit.SECONDS);
+    }
+
+    private boolean readAllHasSufficientNodes()
+    {
+        return RangeCommands.sufficientLiveNodesForSelectStar(SystemDistributedKeyspace.PartitionDenylistTable,
+                                                              DatabaseDescriptor.getDenylistConsistencyLevel());
+    }
+
+    /** Helper method as we need to both build cache on initial init but also on reload of cache contents and params */
+    private LoadingCache<TableId, DenylistEntry> buildEmptyCache()
+    {
+        return CacheBuilder.newBuilder()
+                           .refreshAfterWrite(DatabaseDescriptor.getDenylistRefreshSeconds(), TimeUnit.SECONDS)
+                           .build(new CacheLoader<TableId, DenylistEntry>()
+                           {
+                               @Override
+                               public DenylistEntry load(final TableId tid) throws Exception
+                               {
+                                   return getDenylistForTable(tid);
+                               }
+
+                               @Override
+                               public ListenableFuture<DenylistEntry> reload(final TableId tid, final DenylistEntry oldValue)
+                               {
+                                   ListenableFutureTask<DenylistEntry> task = ListenableFutureTask.create(new Callable<DenylistEntry>()
+                                   {
+                                       @Override
+                                       public DenylistEntry call()
+                                       {
+                                           final DenylistEntry newEntry = getDenylistForTable(tid);
+                                           if (newEntry != null)
+                                               return newEntry;
+                                           if (oldValue != null)
+                                               return oldValue;
+                                           return new DenylistEntry();
+                                       }
+                                   });
+                                   executor.execute(task);
+                                   return task;
+                               }
+                           });
+    }
+
+    /**
+     * We need to fully rebuild a new cache to accommodate deleting items from the denylist and potentially shrinking
+     * the max allowable size in the list. We do not serve queries out of this denylist until it is populated
+     * so as not to introduce a window of having a partially filled cache allow denylisted entries.
+     */
+    public boolean load()
+    {
+        final long start = System.currentTimeMillis();
```

Comment:
The same regarding `Clock` usage.

Reply:
? Don't understand this comment. Elaborate? :)